### PR TITLE
document `#![feature(const_generics_defaults)]`

### PR DESCRIPTION
**Pull request by BoxyUwU**
[stabilisation report](https://github.com/rust-lang/rust/pull/90207#issuecomment-958843546) for this feautre

cc @rust-lang/project-const-generics

### CODE REVIEW

File: src/items/generics.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ fn foo<'a, T>() {}
 trait A<U> {}
 struct Ref<'a, T> where T: 'a { r: &'a T }
 struct InnerArray<T, const N: usize>([T; N]);
+struct Uwu<const N: bool, U>(U);
```
Comment (joshtriplett) [inline src/items/generics.md:34]:
```suggestion
struct EitherOrderWorks<const N: bool, U>(U);
```

No issue with the pwayfuwness, just twying to document what this example is twying to show. :smiley_cat:

**Review by ehuss — COMMENTED** [review]
Thanks for the PR!  There seems to be some information missing, and I was wondering if you could incorporate some more?  For example, some of the following are things that I've noticed:

* What are the restrictions for the expression in the default?
* Restrictions on the type of the default expression.  I believe it must be the same type as the declaration, but can there be any coercion? Is it essentially a [constant expression](https://doc.rust-lang.org/nightly/reference/const_eval.html#constant-expressions)? 

And anything else you can think of?

I understand that default generic parameters are not documented at all (#24). Although it would be great to have that, I think it's fine to not include it now.

Also, it can usually be helpful to provide some links to more information in the PR.  I found the [stabilization report](https://github.com/rust-lang/rust/pull/90207#issuecomment-958843546) which is very useful.

File: src/items/generics.md (hunk 1 above)
Comment (yaahc) [reply src/items/generics.md:34]: Might I suggest `EithewOwdewWowks`

**Review by ehuss — APPROVED** [review]
I'm going to go ahead and merge as-is. However, I would appreciate if anyone can help fill out the details more.  I opened #1159 to track the rest of this.

**Pull request merged**